fastlane/lib/fastlane/plugins/plugin_manager.rb

```diff
@@ -0,0 +1,338 @@
+
+      if gemfile_content.to_s.length > 0
+        UI.important("fastlane will modify your existing Gemfile at path '#{path_to_gemfile}'")
+      else
+        UI.important("fastlane will create a new Gemfile at path '#{path_to_gemfile}'")
+      end
+
+      UI.important("This change is neccessary for fastlane plugins to work")
+
+      unless UI.confirm("Should fastlane modify the Gemfile at path '#{path_to_gemfile}' for you?")
+        UI.important("Please add the following code to '#{path_to_gemfile}':")
+        puts ""
+        puts self.class.code_to_attach.magenta # we use `puts` instead of `UI` to make it easier to copy and paste
+        UI.user_error!("Please update '#{path_to_gemfile} and run fastlane again")
+      end
+
+      attach_plugins_to_gemfile!(path_to_gemfile)
+      UI.success("Successfully modified '#{path_to_gemfile}'")
+    end
+
+    # The code required to load the Plugins file
+    def self.code_to_attach
+      if FastlaneFolder.path
+        fastlane_folder_name = File.basename(FastlaneFolder.path)
+      else
+        fastlane_folder_name = "fastlane"
+      end
+      "plugins_path = File.join(File.dirname(__FILE__), '#{fastlane_folder_name}', '#{PluginManager::PLUGINFILE_NAME}')\n" \
+      "eval(File.read(plugins_path), binding) if File.exist?(plugins_path)"
+    end
+
+    # Makes sure, the user's Gemfile actually loads the Plugins file
+    def plugins_attached?
+      gemfile_path && gemfile_content.include?(self.class.code_to_attach)
+    end
+
+    def ensure_plugins_attached!
+      return if plugins_attached?
+      self.setup
+    end
+
+    #####################################################
+    # @!group Requiring the plugins
+    #####################################################
+
+    # Iterate over all available plugins
+    # which follow the naming convention
+    #   fastlane-plugin-[plugin_name]
+    # This will make sure to load the action
+    # and all its helpers
+    def self.load_plugins
+      UI.verbose("Checking if there are any plugins that should be loaded...")
+
+      loaded_plugins = false
+      Gem::Specification.each do |current_gem|
+        gem_name = current_gem.name
+        next unless gem_name.start_with?(PluginManager.plugin_prefix)
+
+        UI.verbose("Loading '#{gem_name}' plugin")
+        begin
+          require gem_name.tr("-", "/") # from "fastlane-plugin-xcversion" to "fastlane/plugin/xcversion"
+          store_plugin_reference(gem_name)
+          loaded_plugins = true
+        rescue => ex
+          UI.error("Error loading plugin '#{gem_name}': #{ex}")
+
+          # We'll still add it to the table, to make the error
+          # much more visible and obvious
+          self.plugin_references[gem_name] = {
+            version_number: Fastlane::ActionCollector.determine_version(gem_name),
+            actions: []
+          }
+        end
+      end
+
+      if !loaded_plugins && PluginManager.new.pluginfile_content.to_s.include?(plugin_prefix)
```

Comment:
That's fixed with https://github.com/fastlane/fastlane/pull/4927/files#diff-d799ecdf08be1a9f3433cc543c2d22ffR276 👍